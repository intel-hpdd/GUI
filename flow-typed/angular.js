// @flow

declare module angular {
  declare class $scope {
    $new():$scope;
    $digest():void;
  }
  declare type $scopeT = $scope;
  declare class Module {
    provider(name:string, providerType:Function):Module;
    factory(name:string, providerFunction:Function):Module;
    controller(name:string, constructor:Function):Module;
    service(name:string, constructor:Function):Module;
    config(configFn:Function):Module;
    run(initializationFn:Function):Module;
    directive(name:string, directiveFactory:Function):Module;
    component(name:string, options:Object):Module;
    filter(name:string, filterFactory:Function):Module;
    value(name:string, object:any):Module;
    constant(name:string, object:any):Module;
    name:string;
  }
  declare type $compileT = (el:(string | HTMLElement)) => (scope:$scope) => HTMLElement[];
  declare module.exports: {
    module(name:string, dependencies: string[]):Module;
    bootstrap(element:(HTMLElement | Document), modules:string[], config:Object):void;
    merge(...rest:Object[]):Object;
    element(el:(string | HTMLElement)): {
      clone: () => HTMLElement
    };
  }
}
