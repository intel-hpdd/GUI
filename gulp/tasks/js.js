//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
