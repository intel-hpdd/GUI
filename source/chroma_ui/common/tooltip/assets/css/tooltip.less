.help {
  position: relative;

  .tooltip .tooltip-inner {
    width: 150px;
  }
}

.help-tooltip {
  display: inline-block;
}

.tooltip {
  display: none;

  &.in {
    display: block;
  }
}