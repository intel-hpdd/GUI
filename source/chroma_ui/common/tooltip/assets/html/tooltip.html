<div class="tooltip" ng-class="[placement, in]">
  <div class="tooltip-arrow"></div>
  <div class="tooltip-inner" ng-transclude></div>
</div>
