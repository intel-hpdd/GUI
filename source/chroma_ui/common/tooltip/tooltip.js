//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2014 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.


(function () {
  'use strict';

  angular.module('iml-tooltip')
    .directive('imlTooltip', ['position', '$timeout', '$$rAF', 'strategies', imlTooltip])
    .directive('helpTooltip', ['help', helpTooltip]);

  function imlTooltip(position, $timeout, $$rAF, strategies) {
    return {
      scope: {
        toggle: '=?',
        direction: '@'
      },
      restrict: 'E',
      transclude: true,
      replace: true,
      templateUrl: 'common/tooltip/assets/html/tooltip.html',
      link: function link(scope, jqElement) {
        var deregister;

        var jqPreviousEl = jqElement.prev();

        var positioner = position.positioner(jqElement[0]);

        // look at the parent's previous sibling
        if (!jqPreviousEl.length)
          jqPreviousEl = jqElement.parent().prev();

        if (!jqPreviousEl.length)
          throw new Error('Previous element not found for tooltip!');


        var directions = _.values(position.DIRECTIONS),
          directionsJoined = directions.join(' ');

        directions.splice(directions.indexOf(scope.direction), 1);
        directions.unshift(scope.direction);

        scope.$on('$destroy', function onDestroy() {
          jqPreviousEl = null;
          deregister();
        });

        if (!scope.hasOwnProperty('toggle')) {
          deregister = strategies(jqPreviousEl, scope, {
            show: turnThenShow,
            hide: hide
          });
        } else {
          deregister = scope.$watch('toggle', function (newValue) {
            if (newValue)
              turnThenShow();
            else
              hide();
          }, true);
        }

        scope.$watch(function setWatch () {
            return jqElement.html();
          },
          function handleChange (newValue, oldValue) {
            if (newValue !== oldValue) {
              $$rAF(recalculate);
            }
          });

        /**
         * Figures out the placement of the popover and applies it to the element's style.
         */
        function recalculate () {
          jqElement.css('min-width', '');
          jqElement.css(position.position(scope.direction, positioner));
        }

        function turnThenShow() {
          $timeout(show);
        }

        function show() {
          setPosition();

          angular.element(position.$window).on('resize', throttledResize);
        }

        function hide() {
          delete scope.in;

          angular.element(position.$window).off('resize', throttledResize);
        }

        var throttledResize = _.throttle(function() {
          if (!jqElement.hasClass('in')) return;

          setPosition();
          scope.$digest();
        }, 500);

        /**
         * Sets the position of the tooltip by placing a clone until a match is found.
         */
        function setPosition() {
          var clone = jqElement[0].cloneNode(true),
            jqClone = angular.element(clone),
            clonePosition = position.positioner(clone),
            windowPosition = position.positioner(position.$window);

          jqClone.css('display', 'block');

          // place the clone.
          jqElement.after(clone);

          // check where clone fits.
          directions.some(function (direction) {
            delete scope.in;

            jqClone.removeClass(directionsJoined).addClass(direction);

            var calculatedClonePosition = position.position(direction, clonePosition);

            _.extend(clone.style, calculatedClonePosition);

            var fits = !position.overflows(direction, windowPosition, clonePosition);

            if (fits) {
              scope.placement = direction;
              scope.in = 'in';
              _.extend(jqElement[0].style, calculatedClonePosition);
            }

            return fits;
          });

          //destroy the clone
          jqClone.remove();
          clone = null;
          jqClone = null;
        }
      }
    };
  }

  function helpTooltip (help) {
    return {
      scope: {
        toggle: '=?',
        topic: '@',
        direction: '@'
      },
      restrict: 'E',
      replace: true,
      templateUrl: 'common/tooltip/assets/html/help-tooltip.html',
      link: function link(scope) {
        scope.message = help.get(scope.topic);

        if (scope.hasOwnProperty('toggle'))
          scope.hasToggle = true;
      }
    };
  }
}());
