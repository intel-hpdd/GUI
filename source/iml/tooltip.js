// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import Inferno from 'inferno';

export type directionsT = 'left' | 'right' | 'top' | 'bottom';

export type sizesT = '' | 'xsmall' | 'small' | 'medium' | 'large';

type TooltipProps = {
  message?: string,
  direction: directionsT,
  size?: sizesT,
  moreClasses?: string[]
};

export default ({ message, direction, size = '', moreClasses = [] }: TooltipProps) => {
  if (!message) return;

  return (
    <div className={`tooltip inferno-tt ${direction} ${size} ${moreClasses.join(' ')}`}>
      <div class="tooltip-arrow" />
      <div class="tooltip-inner">
        <span>{message}</span>
      </div>
    </div>
  );
};
