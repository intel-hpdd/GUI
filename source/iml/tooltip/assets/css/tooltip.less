.tooltip-container {
  position: relative;

  &.tooltip-hover:hover .tooltip.inferno-tt {
    display: block;
    opacity: .9;
    width: 150px;
  }
}

.has-error {
  > .tooltip.inferno-tt {
    display: block;
    opacity: .9;
  }
}

.tooltip.inferno-tt {
  &.right {
    left: 100%;
    bottom: 50%;
    transform: translateY(50%);
  }

  &.left {
    bottom: 50%;
    transform: translate(-100%, 50%);
  }

  &.bottom {
    transform: translateX(50%);
    right: 50%;
  }

  &.top {
    transform: translate(50%, -100%);
    top: 0;
    right: 50%;
  }
}

.tooltip {
  display: none;

  &.in {
    display: block;
  }
}
