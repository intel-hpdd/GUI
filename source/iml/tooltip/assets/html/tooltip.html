<div class="tooltip inferno-tt" ng-class="[direction, in]">
  <div class="tooltip-arrow"></div>
  <div class="tooltip-inner" ng-transclude></div>
</div>
