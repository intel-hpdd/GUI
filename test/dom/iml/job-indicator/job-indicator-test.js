// @flow

import angular from "../../../angular-mock-setup.js";
import jobIndicator from "../../../../source/iml/job-indicator/job-indicator.js";
import imlPopover from "../../../../source/iml/iml-popover.js";
import { imlTooltip } from "../../../../source/iml/tooltip/tooltip.js";
import Position from "../../../../source/iml/position.js";

describe("job indicator", () => {
  let $rootScope, $scope, $timeout, el;

  beforeEach(
    angular.mock.module(($provide, $compileProvider) => {
      $provide.service("position", Position);
      $compileProvider.component("jobIndicator", jobIndicator);
      $compileProvider.directive("imlTooltip", imlTooltip);
      $compileProvider.directive("imlPopover", imlPopover);
    })
  );

  describe("basic state", () => {
    beforeEach(
      angular.mock.inject(function(_$rootScope_, $compile, _$timeout_) {
        $rootScope = _$rootScope_;
        $timeout = _$timeout_;

        const template = '<job-indicator record-id="id" content-type-id="contentTypeId" locks="locks" />';

        $scope = $rootScope.$new();
        $scope.id = 1;
        $scope.contentTypeId = 98;
        $scope.locks = [];

        el = $compile(template)($scope)[0];
        $scope.$digest();
      })
    );

    it("should render basic state", () => {
      expect(el).toMatchSnapshot();
    });
  });

  describe("with write locks", () => {
    let btn;
    beforeEach(
      angular.mock.inject(function(_$rootScope_, $compile, _$timeout_) {
        $rootScope = _$rootScope_;
        $timeout = _$timeout_;

        const template = '<job-indicator record-id="id" content-type-id="contentTypeId" locks="locks" />';

        $scope = $rootScope.$new();
        $scope.id = 1;
        $scope.contentTypeId = 98;
        $scope.locks = [
          {
            action: "add",
            content_type_id: 98,
            description: "Some write lock",
            item_id: 1,
            job_id: 7,
            lock_type: "write"
          }
        ];

        el = $compile(template)($scope)[0];
        $scope.$digest();

        btn = el.querySelector(".activate-popover");
      })
    );

    it("should render hover state", () => {
      btn.dispatchEvent(new MouseEvent("hover"));
      $timeout.flush();

      expect(el).toMatchSnapshot();
    });

    it("should render popover state", () => {
      btn.click();
      $timeout.flush();

      expect(el).toMatchSnapshot();
    });
  });

  describe("with read locks", () => {
    let btn;
    beforeEach(
      angular.mock.inject(function(_$rootScope_, $compile, _$timeout_) {
        $rootScope = _$rootScope_;
        $timeout = _$timeout_;

        const template = '<job-indicator record-id="id" content-type-id="contentTypeId" locks="locks" />';

        $scope = $rootScope.$new();
        $scope.id = 1;
        $scope.contentTypeId = 98;
        $scope.locks = [
          {
            action: "add",
            content_type_id: 98,
            description: "Some read lock",
            item_id: 1,
            job_id: 7,
            lock_type: "read"
          }
        ];

        el = $compile(template)($scope)[0];
        $scope.$digest();

        btn = el.querySelector(".activate-popover");
      })
    );

    it("should render hover state", () => {
      btn.dispatchEvent(new MouseEvent("hover"));
      $timeout.flush();

      expect(el).toMatchSnapshot();
    });

    it("should render popover state", () => {
      btn.click();
      $timeout.flush();

      expect(el).toMatchSnapshot();
    });
  });

  describe("with read and write locks", () => {
    let btn;
    beforeEach(
      angular.mock.inject(function(_$rootScope_, $compile, _$timeout_) {
        $rootScope = _$rootScope_;
        $timeout = _$timeout_;

        const template = '<job-indicator record-id="id" content-type-id="contentTypeId" locks="locks" />';

        $scope = $rootScope.$new();
        $scope.id = 1;
        $scope.contentTypeId = 98;
        $scope.locks = [];

        el = $compile(template)($scope)[0];
        $scope.$digest();

        $scope.locks = [
          {
            action: "add",
            content_type_id: 98,
            description: "Some write lock",
            item_id: 1,
            job_id: 2,
            lock_type: "write"
          },
          {
            action: "add",
            content_type_id: 98,
            description: "Some read lock",
            item_id: 1,
            job_id: 3,
            lock_type: "read"
          }
        ];
        $scope.$digest();

        btn = el.querySelector(".activate-popover");
      })
    );

    it("should render hover state", () => {
      btn.dispatchEvent(new MouseEvent("hover"));
      $timeout.flush();

      expect(el).toMatchSnapshot();
    });

    it("should render popover state", () => {
      btn.click();
      $timeout.flush();

      expect(el).toMatchSnapshot();
    });
  });

  describe("triggering a change without locks", () => {
    beforeEach(
      angular.mock.inject(function(_$rootScope_, $compile, _$timeout_) {
        $rootScope = _$rootScope_;
        $timeout = _$timeout_;

        const template = '<job-indicator record-id="id" content-type-id="contentTypeId" locks="locks" />';

        $scope = $rootScope.$new();
        $scope.id = 1;
        $scope.contentTypeId = 98;
        $scope.locks = [
          {
            action: "add",
            content_type_id: 98,
            description: "Some read lock",
            item_id: 1,
            job_id: 7,
            lock_type: "read"
          }
        ];

        el = $compile(template)($scope)[0];
        $scope.$digest();
      })
    );

    it("should not update the messages", () => {
      $scope.locks = null;
      $scope.$digest();

      expect(el).toMatchSnapshot();
    });
  });
});
