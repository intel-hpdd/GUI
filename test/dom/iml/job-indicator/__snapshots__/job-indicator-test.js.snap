// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`job indicator basic state should render basic state 1`] = `
<job-indicator
  class="ng-scope ng-isolate-scope"
  content-type-id="contentTypeId"
  locks="locks"
  record-id="id"
>
  <span
    class="job-status"
  >
    
    
    <i
      class="fa fa-lock tooltip-container tooltip-hover activate-popover ng-hide"
      ng-show="$ctrl.shouldShowLockIcon()"
    >
      
       
      <div
        class="tooltip inferno-tt ng-isolate-scope right large"
        direction="right"
        ng-class="[direction, in, size]"
        size="'large'"
      >
        
  
        <div
          class="tooltip-arrow"
        />
        
  
        <div
          class="tooltip-inner"
          ng-transclude=""
        >
          
         
          <span
            class="ng-binding ng-scope"
          >
            
          </span>
          
       
        </div>
        

      </div>
      
    
    </i>
    
  
    <iml-popover
      class="ng-isolate-scope"
      on-toggle="$ctrl.onToggle(state)"
      placement="bottom"
      title="Job Operations"
    />
    

  </span>
</job-indicator>
`;

exports[`job indicator triggering a change without locks should not update the messages 1`] = `
<job-indicator
  class="ng-scope ng-isolate-scope"
  content-type-id="contentTypeId"
  locks="locks"
  record-id="id"
>
  <span
    class="job-status"
  >
    
    
    <i
      class="fa fa-lock tooltip-container tooltip-hover activate-popover"
      ng-show="$ctrl.shouldShowLockIcon()"
    >
      
       
      <div
        class="tooltip inferno-tt ng-isolate-scope right large"
        direction="right"
        ng-class="[direction, in, size]"
        size="'large'"
      >
        
  
        <div
          class="tooltip-arrow"
        />
        
  
        <div
          class="tooltip-inner"
          ng-transclude=""
        >
          
         
          <span
            class="ng-binding ng-scope"
          >
            Locked by 1 pending operation. Click to review details.
          </span>
          
       
        </div>
        

      </div>
      
    
    </i>
    
  
    <iml-popover
      class="ng-isolate-scope"
      on-toggle="$ctrl.onToggle(state)"
      placement="bottom"
      title="Job Operations"
    />
    

  </span>
</job-indicator>
`;

exports[`job indicator with read and write locks should render hover state 1`] = `
<job-indicator
  class="ng-scope ng-isolate-scope"
  content-type-id="contentTypeId"
  locks="locks"
  record-id="id"
>
  <span
    class="job-status"
  >
    
    
    <i
      class="fa fa-lock tooltip-container tooltip-hover activate-popover"
      ng-show="$ctrl.shouldShowLockIcon()"
    >
      
       
      <div
        class="tooltip inferno-tt ng-isolate-scope right large"
        direction="right"
        ng-class="[direction, in, size]"
        size="'large'"
      >
        
  
        <div
          class="tooltip-arrow"
        />
        
  
        <div
          class="tooltip-inner"
          ng-transclude=""
        >
          
         
          <span
            class="ng-binding ng-scope"
          >
            There is 1 ongoing write lock operation and 1 pending read lock operation. Click to review details.
          </span>
          
       
        </div>
        

      </div>
      
    
    </i>
    
  
    <iml-popover
      class="ng-isolate-scope"
      on-toggle="$ctrl.onToggle(state)"
      placement="bottom"
      title="Job Operations"
    />
    

  </span>
</job-indicator>
`;

exports[`job indicator with read and write locks should render popover state 1`] = `
<job-indicator
  class="ng-scope ng-isolate-scope"
  content-type-id="contentTypeId"
  locks="locks"
  record-id="id"
>
  <span
    class="job-status"
  >
    
    
    <i
      class="fa fa-lock tooltip-container tooltip-hover activate-popover"
      ng-show="$ctrl.shouldShowLockIcon()"
    >
      
       
      <div
        class="tooltip inferno-tt ng-isolate-scope right large"
        direction="right"
        ng-class="[direction, in, size]"
        size="'large'"
      >
        
  
        <div
          class="tooltip-arrow"
        />
        
  
        <div
          class="tooltip-inner"
          ng-transclude=""
        >
          
         
          <span
            class="ng-binding ng-scope"
          >
            There is 1 ongoing write lock operation and 1 pending read lock operation. Click to review details.
          </span>
          
       
        </div>
        

      </div>
      
    
    </i>
    
  
    <div
      class="popover fade bottom in"
      ng-class="{in: open}"
      style="display: block; top: 100%; min-width: 0px;"
    >
        
      <div
        class="arrow"
      />
        
      <h3
        class="popover-title ng-binding"
        ng-bind="title"
        ng-show="title"
      >
        Job Operations
      </h3>
        
      <div
        class="popover-content"
      >
        
    
        <accordion
          class="ng-scope"
          close-others="$ctrl.closeOthers"
        >
          
      
          <!-- ngIf: $ctrl.writeMessages.length &gt; 0 ||
      $ctrl.writeMessageDifference.length &gt; 0 -->
          <accordion-group
            class="ng-scope"
            heading="Write Operations"
            is-open="$ctrl.openWrite"
            ng-if="$ctrl.writeMessages.length > 0 ||
      $ctrl.writeMessageDifference.length > 0"
          >
            
        
            <ul>
              
          
              <!-- ngRepeat: message in $ctrl.writeMessages -->
              <li
                class="ng-binding ng-scope"
                ng-repeat="message in $ctrl.writeMessages"
              >
                Some write lock
              </li>
              <!-- end ngRepeat: message in $ctrl.writeMessages -->
              
          
              <!-- ngRepeat: message in $ctrl.writeMessageDifference -->
              
        
            </ul>
            
      
          </accordion-group>
          <!-- end ngIf: $ctrl.writeMessages.length &gt; 0 ||
      $ctrl.writeMessageDifference.length &gt; 0 -->
          
      
          <!-- ngIf: $ctrl.readMessages.length &gt; 0 ||
      $ctrl.readMessageDifference -->
          <accordion-group
            class="ng-scope"
            heading="Read Operations"
            is-open="$ctrl.openRead"
            ng-if="$ctrl.readMessages.length > 0 ||
      $ctrl.readMessageDifference"
          >
            
        
            <ul>
              
          
              <!-- ngRepeat: message in $ctrl.readMessages -->
              <li
                class="ng-binding ng-scope"
                ng-repeat="message in $ctrl.readMessages"
              >
                Some read lock
              </li>
              <!-- end ngRepeat: message in $ctrl.readMessages -->
              
          
              <!-- ngRepeat: message in $ctrl.readMessageDifference -->
              
        
            </ul>
            
      
          </accordion-group>
          <!-- end ngIf: $ctrl.readMessages.length &gt; 0 ||
      $ctrl.readMessageDifference -->
          
    
        </accordion>
        
  
      </div>
        
    </div>
    <iml-popover
      class="ng-isolate-scope"
      on-toggle="$ctrl.onToggle(state)"
      placement="bottom"
      title="Job Operations"
    />
    

  </span>
</job-indicator>
`;

exports[`job indicator with read locks should render hover state 1`] = `
<job-indicator
  class="ng-scope ng-isolate-scope"
  content-type-id="contentTypeId"
  locks="locks"
  record-id="id"
>
  <span
    class="job-status"
  >
    
    
    <i
      class="fa fa-lock tooltip-container tooltip-hover activate-popover"
      ng-show="$ctrl.shouldShowLockIcon()"
    >
      
       
      <div
        class="tooltip inferno-tt ng-isolate-scope right large"
        direction="right"
        ng-class="[direction, in, size]"
        size="'large'"
      >
        
  
        <div
          class="tooltip-arrow"
        />
        
  
        <div
          class="tooltip-inner"
          ng-transclude=""
        >
          
         
          <span
            class="ng-binding ng-scope"
          >
            Locked by 1 pending operation. Click to review details.
          </span>
          
       
        </div>
        

      </div>
      
    
    </i>
    
  
    <iml-popover
      class="ng-isolate-scope"
      on-toggle="$ctrl.onToggle(state)"
      placement="bottom"
      title="Job Operations"
    />
    

  </span>
</job-indicator>
`;

exports[`job indicator with read locks should render popover state 1`] = `
<job-indicator
  class="ng-scope ng-isolate-scope"
  content-type-id="contentTypeId"
  locks="locks"
  record-id="id"
>
  <span
    class="job-status"
  >
    
    
    <i
      class="fa fa-lock tooltip-container tooltip-hover activate-popover"
      ng-show="$ctrl.shouldShowLockIcon()"
    >
      
       
      <div
        class="tooltip inferno-tt ng-isolate-scope right large"
        direction="right"
        ng-class="[direction, in, size]"
        size="'large'"
      >
        
  
        <div
          class="tooltip-arrow"
        />
        
  
        <div
          class="tooltip-inner"
          ng-transclude=""
        >
          
         
          <span
            class="ng-binding ng-scope"
          >
            Locked by 1 pending operation. Click to review details.
          </span>
          
       
        </div>
        

      </div>
      
    
    </i>
    
  
    <div
      class="popover fade bottom in"
      ng-class="{in: open}"
      style="display: block; top: 100%; min-width: 0px;"
    >
        
      <div
        class="arrow"
      />
        
      <h3
        class="popover-title ng-binding"
        ng-bind="title"
        ng-show="title"
      >
        Job Operations
      </h3>
        
      <div
        class="popover-content"
      >
        
    
        <accordion
          class="ng-scope"
          close-others="$ctrl.closeOthers"
        >
          
      
          <!-- ngIf: $ctrl.writeMessages.length &gt; 0 ||
      $ctrl.writeMessageDifference.length &gt; 0 -->
          
      
          <!-- ngIf: $ctrl.readMessages.length &gt; 0 ||
      $ctrl.readMessageDifference -->
          <accordion-group
            class="ng-scope"
            heading="Read Operations"
            is-open="$ctrl.openRead"
            ng-if="$ctrl.readMessages.length > 0 ||
      $ctrl.readMessageDifference"
          >
            
        
            <ul>
              
          
              <!-- ngRepeat: message in $ctrl.readMessages -->
              <li
                class="ng-binding ng-scope"
                ng-repeat="message in $ctrl.readMessages"
              >
                Some read lock
              </li>
              <!-- end ngRepeat: message in $ctrl.readMessages -->
              
          
              <!-- ngRepeat: message in $ctrl.readMessageDifference -->
              
        
            </ul>
            
      
          </accordion-group>
          <!-- end ngIf: $ctrl.readMessages.length &gt; 0 ||
      $ctrl.readMessageDifference -->
          
    
        </accordion>
        
  
      </div>
        
    </div>
    <iml-popover
      class="ng-isolate-scope"
      on-toggle="$ctrl.onToggle(state)"
      placement="bottom"
      title="Job Operations"
    />
    

  </span>
</job-indicator>
`;

exports[`job indicator with write locks should render hover state 1`] = `
<job-indicator
  class="ng-scope ng-isolate-scope"
  content-type-id="contentTypeId"
  locks="locks"
  record-id="id"
>
  <span
    class="job-status"
  >
    
    
    <i
      class="fa fa-lock tooltip-container tooltip-hover activate-popover"
      ng-show="$ctrl.shouldShowLockIcon()"
    >
      
       
      <div
        class="tooltip inferno-tt ng-isolate-scope right large"
        direction="right"
        ng-class="[direction, in, size]"
        size="'large'"
      >
        
  
        <div
          class="tooltip-arrow"
        />
        
  
        <div
          class="tooltip-inner"
          ng-transclude=""
        >
          
         
          <span
            class="ng-binding ng-scope"
          >
            1 ongoing write lock operation. Click to review details.
          </span>
          
       
        </div>
        

      </div>
      
    
    </i>
    
  
    <iml-popover
      class="ng-isolate-scope"
      on-toggle="$ctrl.onToggle(state)"
      placement="bottom"
      title="Job Operations"
    />
    

  </span>
</job-indicator>
`;

exports[`job indicator with write locks should render popover state 1`] = `
<job-indicator
  class="ng-scope ng-isolate-scope"
  content-type-id="contentTypeId"
  locks="locks"
  record-id="id"
>
  <span
    class="job-status"
  >
    
    
    <i
      class="fa fa-lock tooltip-container tooltip-hover activate-popover"
      ng-show="$ctrl.shouldShowLockIcon()"
    >
      
       
      <div
        class="tooltip inferno-tt ng-isolate-scope right large"
        direction="right"
        ng-class="[direction, in, size]"
        size="'large'"
      >
        
  
        <div
          class="tooltip-arrow"
        />
        
  
        <div
          class="tooltip-inner"
          ng-transclude=""
        >
          
         
          <span
            class="ng-binding ng-scope"
          >
            1 ongoing write lock operation. Click to review details.
          </span>
          
       
        </div>
        

      </div>
      
    
    </i>
    
  
    <div
      class="popover fade bottom in"
      ng-class="{in: open}"
      style="display: block; top: 100%; min-width: 0px;"
    >
        
      <div
        class="arrow"
      />
        
      <h3
        class="popover-title ng-binding"
        ng-bind="title"
        ng-show="title"
      >
        Job Operations
      </h3>
        
      <div
        class="popover-content"
      >
        
    
        <accordion
          class="ng-scope"
          close-others="$ctrl.closeOthers"
        >
          
      
          <!-- ngIf: $ctrl.writeMessages.length &gt; 0 ||
      $ctrl.writeMessageDifference.length &gt; 0 -->
          <accordion-group
            class="ng-scope"
            heading="Write Operations"
            is-open="$ctrl.openWrite"
            ng-if="$ctrl.writeMessages.length > 0 ||
      $ctrl.writeMessageDifference.length > 0"
          >
            
        
            <ul>
              
          
              <!-- ngRepeat: message in $ctrl.writeMessages -->
              <li
                class="ng-binding ng-scope"
                ng-repeat="message in $ctrl.writeMessages"
              >
                Some write lock
              </li>
              <!-- end ngRepeat: message in $ctrl.writeMessages -->
              
          
              <!-- ngRepeat: message in $ctrl.writeMessageDifference -->
              
        
            </ul>
            
      
          </accordion-group>
          <!-- end ngIf: $ctrl.writeMessages.length &gt; 0 ||
      $ctrl.writeMessageDifference.length &gt; 0 -->
          
      
          <!-- ngIf: $ctrl.readMessages.length &gt; 0 ||
      $ctrl.readMessageDifference -->
          <accordion-group
            class="ng-scope"
            heading="Read Operations"
            is-open="$ctrl.openRead"
            ng-if="$ctrl.readMessages.length > 0 ||
      $ctrl.readMessageDifference"
          >
            
        
            <ul>
              
          
              <!-- ngRepeat: message in $ctrl.readMessages -->
              
          
              <!-- ngRepeat: message in $ctrl.readMessageDifference -->
              
        
            </ul>
            
      
          </accordion-group>
          <!-- end ngIf: $ctrl.readMessages.length &gt; 0 ||
      $ctrl.readMessageDifference -->
          
    
        </accordion>
        
  
      </div>
        
    </div>
    <iml-popover
      class="ng-isolate-scope"
      on-toggle="$ctrl.onToggle(state)"
      placement="bottom"
      title="Job Operations"
    />
    

  </span>
</job-indicator>
`;
